src/core/containers/HandleLogin/index.js

```diff
@@ -58,7 +59,14 @@ function createLoadData(dispatch) {
             secure: config.get('cookieSecure'),
             maxAge: config.get('cookieMaxAge'),
           });
-          router.push('/search');
+          let to;
+          try {
+            to = browserBase64Decode(state.split(':')[1]);
```

Comment:
We should add some integration tests to verify that it works as expected. The format is clearly quite rigid and I don't think there's a safe way to add more data to this though.